### PR TITLE
[samples_index] Implement carousel

### LINKED ISSUE #393: [visual sample index] Add Carousel
Design spec:

Screen Shot 2020-03-23 at 2 18 19 PM

### PR DESCRIPTION
**Pull request by cazci**
Fixes #393 

I've implemented an animated carousel using dart and scss. Font awesome icons are used for the left and right arrows. Hope this would solve the issue. :)

A preview is added below for reference,

preview

### CODE REVIEW

File: web/samples_index/lib/src/carousel.dart [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+class Carousel {
+  final bool withArrowKeyControl;
+
+  final container = querySelector('.slider-container');
+  var slides = querySelectorAll('.slider-single');
+
+  var lastSlide;
+  var currentSlide;
+  var prevSlide;
+  var nextSlide;
```
Comment (johnpryan) [inline web/samples_index/lib/src/carousel.dart:16]: nitpick: give these a type declaration

```suggestion
  Element lastSlide;
  Element currentSlide;
  Element prevSlide;
  Element nextSlide;
```

File: web/samples_index/lib/src/carousel.dart [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file
+
+import 'dart:html';
+
+class Carousel {
+  final bool withArrowKeyControl;
+
+  final container = querySelector('.slider-container');
+  var slides = querySelectorAll('.slider-single');
```
Comment (johnpryan) [inline web/samples_index/lib/src/carousel.dart:11]: nit: give these a type declaration:

```suggestion
  final Element container = querySelector('.slider-container');
  List<Element> slides = querySelectorAll('.slider-single');
```

**Review by johnpryan — APPROVED** [review]
Cool, thanks!

**Comment by cazci** [comment]
Thanks for reviewing, @domesticmouse @johnpryan 

I've committed the suggestions :)

**Review by johnpryan — APPROVED** [review]

**Comment by RedBrogdon** [comment]
@johnpryan, any reason not to merge this? I was about to, but thought you might have held off for a reason.

**Comment by johnpryan** [comment]
Nope, looks good.

**Pull request merged**